src/sql-parser/src/ast/defs/value.rs

```diff
@@ -59,6 +62,21 @@ pub enum Value {
     Null,
 }
 
+impl Value {
+    pub fn var_type(&self) -> String {
```

Comment:
I did not realize adding the type column would require this, and I'm pretty sure it's not a good idea. We already have a bunch of value, datum, and type types. I'm not convinced that adding a new type layer into that is what we should be doing, instead of re-using an existing one. It might be reasonable to rethink part of the design doc and use something...type-ier instead of ast::Value. Can we copy the pattern of Session `vars` so we have hardcoded types for each specific variable?

I don't see `mz_system_configuration` in the design doc though, so maybe we can do other stuff. Should we just not have a type column in it? That seems both bad and good. I'm not sure what `mz_system_configuration` is needed for, so it's unclear what its requirements are.

Reply:
I may have misunderstood the design doc. I had assumed that 
> a new table in the system catalog stash

meant a new stash collection and a system table. However, now that I'm re-reading it I don't think that's the case. I thought that a system table would be useful for users to be able to see their limits. I think we should remove `mz_system_configuration` for now and maybe add it back in later. The stash value is in bytes so a migration from one type to another should be fairly easy.

Reply:
I think we should do what Postgres does and just extend the `SHOW ...` to be able to show system configurations.